scrapy/dupefilters.py

```diff
@@ -51,9 +58,26 @@ def __init__(self, path: Optional[str] = None, debug: bool = False) -> None:
             self.fingerprints.update(x.rstrip() for x in self.file)
 
     @classmethod
-    def from_settings(cls: Type[RFPDupeFilterTV], settings: BaseSettings) -> RFPDupeFilterTV:
+    def from_settings(cls: Type[RFPDupeFilterTV], settings: BaseSettings, *, fingerprinter=None) -> RFPDupeFilterTV:
         debug = settings.getbool('DUPEFILTER_DEBUG')
-        return cls(job_dir(settings), debug)
+        try:
+            return cls(job_dir(settings), debug, fingerprinter=fingerprinter)
+        except TypeError:
+            result = cls(job_dir(settings), debug)
+            result.fingerprinter = fingerprinter
+            return result
```

Comment:
Is the idea to support RFPDupeFilter subclasss which don't have fingerprint argument in `__init__` method? If so, would it make sense to show some kind of deprecation warning, and ask users to update the code?